Optional void give a C error

**V version:** V 0.1.27 7a5a4df - 0f251e9
**OS:** Linux

**What did you do?**
```v
fn main() {
	foo() or { panic(err) }
}

fn foo() ?void {
	return error('something')
}
```

**What did you expect to see?**
V compiler allow it or give an error 
 
**What did you see instead?**
```
==================
/home/don/.cache/v/main.tmp.c:1157:9: error: variable or field ‘data’ declared void
 1157 |   void  data;
      |         ^~~~
/home/don/.cache/v/main.tmp.c: In function ‘_STR’:
/home/don/.cache/v/main.tmp.c:1266:14: warning: pointer targets in passing argument 1 of ‘tos2’ differ in signedness [-Wpointer-sign]
 1266 |  return tos2(buf);
      |              ^~~
      |              |
      |              char *
/home/don/.cache/v/main.tmp.c:1009:21: note: expected ‘byteptr’ {aka ‘unsigned char *’} but argument is of type ‘char *’
 1009 | string tos2(byteptr s);
      |             ~~~~~~~~^
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
```

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -323,8 +323,9 @@ fn (g &Gen) base_type(t table.Type) string {
 fn (mut g Gen) register_optional(t table.Type, styp string) {
 	// g.typedefs2.writeln('typedef Option $x;')
 	no_ptr := styp.replace('*', '_ptr')
+	typ := if styp == 'void' { 'void*' } else { styp }
 	g.hotcode_definitions.writeln('typedef struct {
-		$styp  data;
+		$typ  data;
 		string error;
 		int    ecode;
 		bool   ok;

--- a/vlib/v/parser/parse_type.v
+++ b/vlib/v/parser/parse_type.v
@@ -123,7 +123,10 @@ pub fn (mut p Parser) parse_type() table.Type {
 		p.next()
 		p.check(.dot)
 	}
-	mut typ := p.parse_any_type(is_c, is_js, nr_muls > 0)
+	mut typ := table.void_type
+	if p.tok.kind != .lcbr {
+		typ = p.parse_any_type(is_c, is_js, nr_muls > 0)
+	}
 	if is_optional {
 		typ = typ.set_flag(.optional)
 	}

--- a/vlib/v/tests/option_test.v
+++ b/vlib/v/tests/option_test.v
@@ -219,3 +219,16 @@ fn test_optional_void() {
 		return
 	}
 }
+
+
+fn bar() ? {
+	return error('bar error')
+}
+
+fn test_optional_void_only_question() {
+	bar() or {
+		println(err)
+		assert err == 'bar error'
+		return
+	}
+}

--- a/vlib/v/tests/option_test.v
+++ b/vlib/v/tests/option_test.v
@@ -207,3 +207,15 @@ fn test_multi_return_opt() {
 	}
 }
 */
+
+fn foo() ?void {
+	return error('something')
+}
+
+fn test_optional_void() {
+	foo() or {
+		println(err)
+		assert err == 'something'
+		return
+	}
+}

